src/uwtools/config/formats/ini.py

```diff
@@ -63,31 +51,25 @@ def dump(self, path: OptionalPath) -> None:
 
         :param path: Path to dump config to.
         """
-        INIConfig.dump_dict(path, self.data, ns(space=self.space_around_delimiters))
+        self.dump_dict(path, self.data)
 
     @staticmethod
-    def dump_dict(path: OptionalPath, cfg: dict, opts: Optional[ns] = None) -> None:
+    def dump_dict(path: OptionalPath, cfg: dict) -> None:
         """
-        Dumps a provided config dictionary in INI or sh format.
+        Dumps a provided config dictionary in INI format.
 
         :param path: Path to dump config to.
         :param cfg: The in-memory config object to dump.
-        :param space_around_delimiters: Place spaces around delimiters?
         """
         # Configparser adds a newline after each section, presumably to create nice-looking output
         # when an INI contains multiple sections. Unfortunately, it also adds a newline after the
         # _final_ section, resulting in an anomalous trailing newline. To avoid this, write first to
         # memory, then strip the trailing newline.
+
         parser = configparser.ConfigParser()
         s = StringIO()
-        cfgdepth = depth(cfg)
-        assert cfgdepth in (1, 2)  # 1 => .sh, 2 => .ini
-        if cfgdepth == 2:
-            parser.read_dict(cfg)
-            parser.write(s, space_around_delimiters=opts.space if opts else True)
-        else:
-            for key, value in cfg.items():
-                print(f"{key}={value}", file=s)
+        parser.read_dict(cfg)
+        parser.write(s, space_around_delimiters=True)
```

Comment:
`space_around_delimiters` is True by default, so you might simplify to `parser.write(s)` for INI formats.